homeassistant/components/worldclock/sensor.py

```diff
@@ -9,18 +9,24 @@
 from homeassistant.helpers.entity import Entity
 import homeassistant.util.dt as dt_util
 
+CONF_TIME_FORMAT = "time_format"
+CONF_TRIM_ZERO = "trim_leading_zero"
+
 _LOGGER = logging.getLogger(__name__)
 
 DEFAULT_NAME = "Worldclock Sensor"
+DEFAULT_TRIM_ZERO = False
 
 ICON = "mdi:clock"
 
-TIME_STR_FORMAT = "%H:%M"
+DEFAULT_TIME_STR_FORMAT = "%H:%M"
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
     {
         vol.Required(CONF_TIME_ZONE): cv.time_zone,
         vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_TIME_FORMAT, default=DEFAULT_TIME_STR_FORMAT): cv.string,
+        vol.Optional(CONF_TRIM_ZERO, default=DEFAULT_TRIM_ZERO): cv.boolean,
```

Comment:
We shouldn't add trim zero option. You can just change the time format to achieve the format you want?

Reply:
strftime() can take %H for 24 hour and %I for 12 hour but it seems both will prefix 0 for hour less then 10.

Reply:
https://stackoverflow.com/questions/904928/python-strftime-date-without-leading-0

Reply:
Thank you. I will revise the merge and take out trim_zero setting.